notebooks/106-speech-recognition-quantize-wav2vec2/106-speech-recognition-quantize-wav2vec2.ipynb

```diff
@@ -0,0 +1,631 @@
+    "ir_model_bin = onnx_model_path.with_suffix(\".bin\")\n",
+    "\n",
+    "if not ir_model_xml.exists():\n",
+    "    !mo --input_model $onnx_model_path --output_dir $MODEL_DIR --model_name wav2vec2_base --input inputs --input_shape [1,-1] --data_type FP16"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "635f4b0d",
+   "metadata": {
+    "id": "LBbY7c4NsHzT"
+   },
+   "source": [
+    "## Prepare LibriSpeech Dataset\n",
+    "\n",
+    "The code below will download and unpack the archive with 'test-clean' subset of LibriSpeech Dataset"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "43070514",
+   "metadata": {
+    "id": "NN-qRME1a-Sm"
+   },
+   "outputs": [],
+   "source": [
+    "download_file(\"http://openslr.elda.org/resources/12/test-clean.tar.gz\", directory=DATA_DIR, show_progress=True)\n",
+    "\n",
+    "if not os.path.exists(f'{DATA_DIR}/LibriSpeech'):\n",
+    "    with tarfile.open(f\"{DATA_DIR}/test-clean.tar.gz\") as tar:\n",
+    "        tar.extractall(path=DATA_DIR)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "da802acd",
+   "metadata": {
+    "id": "E5hsOsj-0uSc"
+   },
+   "source": [
+    "## Define DataLoader for POT\n",
+    "In this step, we need to define `DataLoader` based on POT API. It will be used to collect statistics for quantization and run model evaluation.\n",
+    "Wav2Vec2 model accepts a raw waveform of the speech signal as input and produces vocabulary class estimations as output.\n",
+    "\n",
+    "Consider increase `samples_limit` to get more precise results, suggested value is `300` or more, it will take longer time to process."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "d2a378cb",
+   "metadata": {
+    "id": "6xnl2PhM0uSn",
+    "tags": [],
+    "test_replace": {
+     "return len(self._ds)": "return 3"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "class LibriSpeechDataLoader(POTDataLoader):\n",
+    "    samples_limit = 50\n",
+    "    # Required methods\n",
+    "    def __init__(self, config):\n",
+    "        \"\"\"Constructor\n",
+    "        :param config: data loader specific config\n",
+    "        \"\"\"\n",
+    "        if not isinstance(config, Dict):\n",
+    "            config = Dict(config)\n",
+    "        super().__init__(config)\n",
+    "        self._data_dir = config[\"data_source\"]\n",
+    "        self._ds = []\n",
+    "        self._prepare_dataset()\n",
+    "\n",
+    "    def __len__(self):\n",
+    "        \"\"\"Returns size of the dataset\"\"\"\n",
+    "        return len(self._ds)\n",
+    "\n",
+    "    def __getitem__(self, index):\n",
+    "        \"\"\"\n",
+    "        Returns annotation, data and metadata at the specified index.\n",
+    "        Possible formats:\n",
+    "        (index, annotation), data\n",
+    "        (index, annotation), data, metadata\n",
+    "        \"\"\"\n",
+    "        return self._ds[index]\n",
```

Comment:
If I were you, I'd rather return a dictionary (`{input_name: input_data}`) instead of NumPy arrays, to avoid the issues with the data shapes and etc.
@AlexKoff88, what do you think? Should it be better to use a dictionary here?

Reply:
It should, especially if to note the model is dynamic in terms of the size of input data so the dict should be safer as we convinced before.